### DISCUSSION TITLE
ESP32 Compile error when USE_I2S_AUDIO is enabled

### DISCUSSION THREAD

**Discussion by 13** [arendst/Tasmota#18258] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
With USE_I2S_AUDIO compiling error:
Tasmota/tasmota/tasmota_xdrv_driver/xdrv_42_0_i2s_audio.ino:40:10: fatal error: AudioFileSourcePROGMEM.h: No such file or directory

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): ESP32 DevKit
- [x] Tasmota binary firmware version number used: Latest GIT
  - [ ] Pre-compiled
  - [x] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Compile with USE_I2S_AUDIO 
Tasmota/tasmota/tasmota_xdrv_driver/xdrv_42_0_i2s_audio.ino:40:10: fatal error: AudioFileSourcePROGMEM.h: No such file or directory

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment, marked as answer]
Your Platformio Setup is not correct. You have to enable (uncomment) in `platformio override.ini` the audio libs
```
lib_extra_dirs           = ${library.lib_extra_dirs}
; *** ESP32 lib. ALWAYS needed for ESP32 !!!
                          lib/libesp32
; *** comment the following line if you dont use LVGL in a Tasmota32 build. Reduces compile time
                          lib/libesp32_lvgl
; *** comment the following line if you dont use ESP32 Audio in a Tasmota32 build. Reduces compile time
;                          lib/libesp32_audio
; *** uncomment the following line if you use Bluetooth or Apple Homekit in a Tasmota32 build. Reduces compile time
;                          lib/libesp32_div
; *** uncomment the following line if you use Epaper driver epidy in your Tasmota32 build. Reduces compile time
;                          lib/libesp32_eink
```

**Reply by 13** [reply]
And lib/lib_audio must be enabled
